### PR TITLE
Fix name conflict

### PR DESCRIPTION
**Pull request by mshima**
Fixes https://github.com/jhipster/generator-jhipster/issues/13142

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by DanielFran** [comment]
Trusting you on this

File: generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs [hunk 1]
```diff
@@ -619,16 +619,16 @@ relationships.forEach((relationship, idx) => {
         <%_ if (relationship.otherRelationship && !relationship.reference.owned) { _%>
         if (this.<%= relationshipFieldNamePlural %> != null) {
             <%_ if (relationship.otherRelationship.reference.collection) { _%>
-            this.<%= relationshipFieldNamePlural %>.forEach(<%= otherEntityName %> -> <%= otherEntityName %>.remove<%= otherEntityRelationshipNameCapitalized %>(this));
+            this.<%= relationshipFieldNamePlural %>.forEach(i -> i.remove<%= otherEntityRelationshipNameCapitalized %>(this));
```
Comment (MathieuAA) [inline generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs:622]: I guess we can find a better name for that, right? ;)

**Review by MathieuAA — CHANGES_REQUESTED** [review]

**Comment by MathieuAA** [comment]
@DanielFran you were too fast for me :)

File: generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs (hunk 1 above)
Comment (mshima) [reply generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs:622]: I was going to name it item, but I could not guarantee it won’t conflict with Item entity.

I am curious about the conflict, a plural name was conflicting with a non plural name.
But the code is broken and I don’t have time to go further, I created this workaround.

Comment (MathieuAA) [reply generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs:622]: Hum, that's interesting indeed!

Comment (mshima) [reply generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs:622]: A quick alternative is to name it `<%= otherEntityName %>Item`.

**Pull request merged**